tests/simulator/amr_convergence/amr_convergence_base.py

```
@@ -0,0 +1,148 @@
+    def check_spatial_order(self, order, Ns, sigma, band):
+        """N-sweep at fixed sigma; assert the composite convergence order."""
+        print(
+            f"\n== {self.name}: spatial convergence, order={order}, "
+            f"sigma={sigma} =="
+        )
+        rows = []
+        for N in Ns:
+            n = self.n_steps(N, sigma)
+            per_level, composite = self.run_amr_case(order, N, n)
+            rows.append((N, composite, per_level))
+            lvls = "  ".join(f"L{l}={e:.3e}" for l, e in sorted(per_level.items()))
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)